src/components/DataTable/utils/DataTableColumnUtils.tsx

```diff
@@ -0,0 +1,58 @@
+
+import {type ColumnDef} from '@tanstack/react-table';
+import type {DataTableProps, ColumnType, CellContent} from '../DataTable.types';
+import {TableCellDate} from '../cells/TableCellDate';
+import {TableCellNumber} from '../cells/TableCellNumber';
+import {TableCellChip} from '../cells/TableCellChip';
+import {TableCellActions} from '../cells/TableCellActions';
+import {TableCellText} from '../cells/TableCellText';
+
+
+const CustomCell = (
+    type: ColumnType,
+    value: string | number | Date | CellContent | null | undefined,
+    align: 'left' | 'center' | 'right',
+    locale?: string
+) => {
+    switch (type) {
+        case 'date':
+            return <TableCellDate value={value as Date} locale={locale}/>;
+        case 'number':
+            return <TableCellNumber value={value as number} locale={locale}/>;
+        case 'badge':
+            return <TableCellChip value={value as string}/>;
+        case 'actions':
+            return <TableCellActions/>;
+        case 'text':
+        default:
+            return <TableCellText value={value as string | CellContent}/>;
+    }
+};
```

Comment:
I think we should directly use the Cell component instead of mapping the component to a string type.